syntax = "proto3";

package notification_center;


// 设置用户Apns请求的参数
message SetUserApnsRequest {
    bytes uid = 1; // 用户身份标识
    string apn = 2; // 用户apn
}

// 发送iOS设备通知
message SendApnsNotificationRequest {
    message AppleNotification {
        message AppleAlert {
            string title = 1;   // 通知标题
            optional string subtitle = 2;   // 通知标题，为空字符串则不发送该字段
            string body = 3;   // 通知标题
        }

        AppleAlert alert = 1;   // Apple Remote Notification
        int32 badge = 2;    // 应用程序图标上徽章中显示的数字。指定0以删除当前徽章（如果有的话），置-1以不发送该字段
        string category = 3;    // 通知的类型，为空字符串则不发送该字段
    }

    bytes uid = 1; // 用户身份标识
    AppleNotification notification = 2;     // 通知信息设置
}